spa/src/app/files/hca-download-file/hca-download-file.component.html

```diff
@@ -1,19 +1,20 @@
-<ng-container *ngIf="(observeFileDownloadStatus() | async) as downloadStatus">
-    <img *ngIf="file.fileFormat != 'matrix' && isDownloadNotStarted(downloadStatus)"
-         src="assets/images/icon/hca-download-black.png"
-         (click)="onRequestFile()">
-
-    <img *ngIf="file.fileFormat == 'matrix'"
-         src="assets/images/icon/noFileType.png">
-
-    <mat-spinner *ngIf="isDownloadInProgress(downloadStatus)"
-                 class="hca-spinner"
-                 [diameter]="14"
-                 [mode]="indeterminate">
-    </mat-spinner>
-    <ng-container *ngIf="isDownloadFailed(downloadStatus)"><div class="error">x</div></ng-container>
-    <img *ngIf="isDownloadComplete(downloadStatus)"
-         src="assets/images/icon/check-black.png">
+<ng-container [ngSwitch]="file.fileFormat === 'matrix'">
+    <ng-container *ngSwitchCase="true">
+        <img src="assets/images/icon/noFileType.png">
+    </ng-container>
+    <ng-container *ngSwitchDefault>
+        <ng-container *ngIf="(observeFileDownloadStatus() | async) as downloadResponse">
+            <img *ngIf="isDownloadNotStarted(downloadResponse)"
+                 src="assets/images/icon/hca-download-black.png" (click)="onRequestFile()">
+            <mat-spinner *ngIf="showSpinner(downloadResponse)"
+                         class="hca-spinner"
+                         [diameter]="14"
+                         [mode]="indeterminate">
+            </mat-spinner>
+            <ng-container *ngIf="isDownloadFailed(downloadResponse)"><div class="error">x</div></ng-container>
+            <img *ngIf="isDownloading(downloadResponse)"
+                 src="assets/images/icon/check-black.png">
+        </ng-container>
+        <a download class="hidden" #download></a>
```

Comment:
So the content disposition header can be embedded into the signed S3 URL. We could try to convince the DSS people to do this during checkout. We could also (very hacky) parse the signed URL returned by DSS for the object key and resign it with the header present.

However, the first thing I propose we try is actually set the `download` attribute to the file name:

```suggestion
        <a download="{{file.fileName}}" class="hidden" #download></a>
```

Reply:
References:

https://davidwalsh.name/download-attribute
https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html
https://github.com/boto/boto3/issues/610#issuecomment-443878760

Reply:
@NoopDog

Reply:
Ok thanks will give this a try.